soccer/src/soccer/planning/planner_node.cpp

```diff
@@ -122,12 +125,14 @@ Trajectory PlannerForRobot::plan_for_robot(const planning::PlanRequest& request)
         // If this planner could possibly plan for this command, try to make
         // a plan.
         if (trajectory.empty() && planner->is_applicable(request.motion_command)) {
+            current_planner_ = planner;
             trajectory = planner->plan(request);
         }
 
         // If it fails, or if the planner was not used, the trajectory will
         // still be empty. Reset the planner.
         if (trajectory.empty()) {
+            current_planner_ = nullptr;
```

Comment:
the `current_planner_` var doesn't actually do anything other than store a ptr to the planner selected in the for-loop above (lines 127-130). Notice there that the trajectory is taken directly from one of the planners in the for-loop.

Reply:
the replanner resetting from null is because it gets a fresh RobotIntent from gameplay every tick. another reason why we need to overhaul gameplay to use actions instead of a blind gameplay->RobotIntent->planning pipeline

Reply:
(for clarification, the ptr is only necessary here for the debug prints below, where I need to get the name of whatever planner is saying it's done)